run.go

```diff
@@ -170,31 +180,30 @@ func Run() orderedmap.OrderedMap[string, orderedmap.OrderedMap[string, TestResul
 					}()
 
 					if afterTestErr != nil {
-						// replace test results with new one marking it failed and with this panic message.
-						old := results[pkg][name]
-						results[pkg][name] = TestResult{
-							Package: old.Package,
-							Name:    old.Name,
-							Passed:  false,
-							Dur:     old.Dur,
-							Msgs: append(old.Msgs, append(testHelperT.msgs, Msg{
-								Msg:   fmt.Sprintf("%v", afterTestErr),
-								Level: LevelError,
-							})...),
-						}
+						pkgAnyFailures = true
+						// update test results marking it failed and with this panic message.
+						r := &pkgResults.Subtests[len(pkgResults.Subtests)-1]
```

Comment:
This pattern is fine but feels awkward. Maybe a function on a typed []TestResult receiver to access last element?

Reply:
I'm not sold on making an opaque type just to avoid slice length calculations in 2 locations